src/storage-transport.ts

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {GaxiosError, GaxiosOptions, Headers} from 'gaxios';
+import {AuthClient, GoogleAuth, GoogleAuthOptions} from 'google-auth-library';
+import {
+  getModuleFormat,
+  getRuntimeTrackingString,
+  getUserAgentString,
+} from './util';
+import {randomUUID} from 'crypto';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import {getPackageJSON} from './package-json-helper.cjs';
+import {GCCL_GCS_CMD_KEY} from './nodejs-common/util';
+import {RetryOptions} from './storage';
+
+export interface StandardStorageQueryParams {
+  alt?: 'json' | 'media';
+  callback?: string;
+  fields?: string;
+  key?: string;
+  prettyPrint?: boolean;
+  quotaUser?: string;
+  userProject?: string;
+}
+
+export interface StorageQueryParameters extends StandardStorageQueryParams {
+  [key: string]: string | number | boolean | undefined;
+}
+
+export interface StorageRequestOptions extends GaxiosOptions {
+  [GCCL_GCS_CMD_KEY]?: string;
+  //interceptors_?: Interceptor[];
+  autoPaginate?: boolean;
+  autoPaginateVal?: boolean;
+  maxRetries?: number;
+  objectMode?: boolean;
+  projectId?: string;
+  queryParameters?: StorageQueryParameters;
+  shouldReturnStream?: boolean;
+}
+
+export interface StorageCallback<T> {
+  (err: GaxiosError<T> | null, data?: T): void;
+}
+
+interface TransportParameters extends Omit<GoogleAuthOptions, 'authClient'> {
+  apiEndpoint: string;
```

Comment:
are these options all copied from somewhere? What's the source and how do we know nothing is missing?

Reply:
Yes, this is extending `GoogleAuthOptions` but omitting `authClient`.